contracts/executor/src/lib.rs

```diff
@@ -0,0 +1,263 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+extern crate alloc;
+use ink_lang as ink;
+
+mod task;
+
+#[allow(clippy::large_enum_variant)]
+#[ink::contract(env = pink_extension::PinkEnvironment)]
+mod index_executor {
+    use alloc::vec::Vec;
+    use index::ensure;
+    use index::prelude::*;
+    use index::prelude::{ChainInfo, Graph};
+    use index::registry::bridge::{AssetPair, Bridge};
+    use index::registry::chain::Chain;
+    use index::registry::dex::{Dex, DexPair};
+    use ink_storage::traits::SpreadAllocate;
+    use ink_storage::Mapping;
+    // use pallet_index::{};
+    use ink_env::call::FromAccountId;
+    use index_registry::{AccountInfo, AccountStatus, RegistryRef};
+
+    use task::{ActivedTaskFetcher, RuningTaskFetcher, Step, TaskClaimer, TaskUploader};
+
+    #[derive(Encode, Decode, Debug, PartialEq, Eq)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        ExecuteFailed,
+    }
+
+    #[ink(storage)]
+    #[derive(SpreadAllocate)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub struct Executor {
+        pub admin: AccountId,
+        pub registry: RegistryRef,
+        pub worker_accounts: Vec<[u8; 32]>,
+        pub executor_account: [u8; 32],
+    }
+
+    impl Default for Executor {
+        fn default() -> Self {
+            Self::new()
+        }
+    }
+
+    impl Executor {
+        #[ink(constructor)]
+        /// Create an Executor entity
+        pub fn new(register: AccountId) -> Self {
+            ink_lang::utils::initialize_contract(|this: &mut Self| {
+                this.admin = Self::env().caller();
+                this.registry = RegistryRef::from_account_id(register);
+                for index in 0..10 {
+                    // TODO: generate private key
+                    this.worker_accounts.push(vec![0; 32])
+                }
+                pink_extension::ext().cache_set(b"alloc", vec![])?;
+            })
+        }
+
+        /// Claim and execute tasks from all supported blockchains. This is a query operation
+        /// that scheduler invokes periodically.
+        /// 
+        /// 
+        /// 1) Perform spcific operations for the runing tasks according to current status.
+        /// 2) Fetch new actived tasks from supported chains and append them to the local runing tasks queue.
+        /// 
+        #[ink::message]
+        pub fn execute(&self) -> Result<(), Error> {
```

Comment:
Consider switching worker and dirty cache problem?

E.g. If worker A has executed some steps and dropped, then the scheduler resume the execution on worker B. Will it be problematic if worker A come back (mainly to avoid double spending)